### DISCUSSION TITLE
PC-SAFT with DFT

### DISCUSSION THREAD

**Discussion by Felip-552** [feos-org/feos#102] [Q&A] [answered]
Hey there! I'm trying to understand the DFT aspect of $\text{FeO}_\text{s}$, specifically when it comes to using PC-SAFT fluids. I'm still pretty new to DFT and learning the basics. I've been having a tough time finding information about how to couple PC-SAFT with DFT – most of what I've found is for LJ fluids and I'm not sure if the same principles apply. 
Do you have any articles on how you implemented this that you could share? Or maybe you could explain the theory behind it? Is it FMT? I'd really appreciate any help you can give me.
Nice work btw! Impressive stuff.

**Comment by prehner** [comment, marked as answer]
Hey thanks for the feedback! The PC-SAFT Helmholtz energy functional is indeed using FMT for the hard sphere reference contribution. The attractive interactions are then modeled using a weighted density approximation. A good place to start is [Sauer and Gross 2015](https://pubs.acs.org/doi/full/10.1021/acs.iecr.6b04551) which describes in detail the development of the weighted density approximation for the dispersion contribution as it is currently implemented in FeOs. Then there are other publications from Joachim Gross' group in Stuttgart about the implementation details (https://www.sciencedirect.com/science/article/pii/S037838121930367X) or applications like droplets (https://aip.scitation.org/doi/full/10.1063/1.5020421), porous media (https://www.sciencedirect.com/science/article/pii/S1387181121003899), or surfactants (https://pubs.acs.org/doi/full/10.1021/acs.iecr.1c00169).

**Reply by Felip-552** [reply]
Thankss for your help! I'll look into the articles.

**Comment by Felip-552** [comment]
Hi, I was wondering if there is a way to determine the excess adsorption in FeOs, since most experimental data is presented as excess adsorption it could be very helpful to determine this magnitude. Or perhaps it could be easier to only determine the volume of the adsorbed phase and use the formula $N_{ads}^{total}=N_{ads}^{excess}+V_a \rho_g$, where $V_a$ is the volume of adsorbate in the adsorbed phase $[\frac{m^3}{g}]$ , and $\rho_g$ is the density of the bulk gas  $[\frac{mol}{m^3}]$  [[1]](https://pubs.acs.org/doi/10.1021/la991159w).

**Reply by prehner** [reply]
You can do exactly that. The volume of the pore can be calculated using `Pore1D.pore_volume` and the density of the bulk phase by `profile.bulk.density`. The units could look confusing at first, because the DFT calculation has no information about specific surface areas or other properties defining the solid. Therefore, the unit of `pore_volume` will depend on the geometry. For a 1D slit pore it is $[m]$, for a spherical pore $[m^3]$. However, it is consistent with the units of `Profile.total_moles` and `Adsorption.total_adsorption` so that the expression that you give can always be evaluated straightforwardly.

The pore volume is calculated using Helium as reference, which is common practice for molecular simulations of porous media.

**Reply by Felip-552** [reply]
Thank you very much, that worked great!

**Comment by cpemjean** [comment]
Hello, I'm also reading about DFT-PC-SAFT, and I have a question: do you utilize an association contribution to the residual intrinsic Helmholtz energy? If yes, is it defined in the same manner as in PC-SAFT?

**Reply by prehner** [reply]
Hello, the PC-SAFT Helmholtz energy functional is completely compatible with the PC-SAFT equation of state, i.e., for a homogeneous bulk phase both give the same result, including the association contribution. The specific Helmholtz energy functional contribution for association is an adaptation of the [Yu and Wu](https://aip.scitation.org/doi/abs/10.1063/1.1463435) model to the PC-SAFT equation of state.

**Reply by cpemjean** [reply]
Thanks for the clarification!
